common/lib/xmodule/xmodule/graders.py

```diff
@@ -10,13 +10,67 @@
 import random
 import sys
 
-from collections import namedtuple
 
 log = logging.getLogger("edx.courseware")
 
-# This is a tuple for holding scores, either from problems or sections.
-# Section either indicates the name of the problem or the name of the section
-Score = namedtuple("Score", "earned possible graded section module_id")
+
+class ScoreBase(object):
+    """
+    Abstract base class for encapsulating fields of values scores.
+    Field common to all scores include:
+        display_name (string) - the display name of the module
+        module_id (UsageKey) - the location of the module
+        graded (boolean) - whether or not this module is graded
+    """
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, graded, display_name, module_id):
+        self.graded = graded
+        self.display_name = display_name
+        self.module_id = module_id
+
+    def __eq__(self, other):
+        if type(other) is type(self):
+            return self.__dict__ == other.__dict__
+        return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        return u"{class_name}({fields})".format(class_name=self.__class__.__name__, fields=self.__dict__)
+
+
+class ProblemScore(ScoreBase):
+    """
+    Encapsulates the fields of a Problem's score.
+    In addition to the fields in ScoreBase, also includes:
+        r_earned (float) - "raw" points "earned" on this problem
+        r_possible (float) - "raw" points "possible" to earn on this problem
+        w_earned = earned (float) - "weighted" value of the points "earned"
+        w_possible = possible (float) - "weighted" "possible" points on this problem
+        weight (float) - weight of this problem
+    """
+    def __init__(self, r_earned, r_possible, w_earned, w_possible, weight, *args, **kwargs):
+        super(ProblemScore, self).__init__(*args, **kwargs)
+        self.r_earned = r_earned
+        self.r_possible = r_possible
+        self.earned = w_earned
+        self.possible = w_possible
+        self.weight = weight
+
+
+class AggregatedScore(ScoreBase):
+    """
+    Encapsulates the fields of a Subsection's score.
+    In addition to the fields in ScoreBase, also includes:
+        tw_earned = earned - total aggregated sum of all weighted earned values
+        tw_possible = possible - total aggregated sum of all possible values
```

Comment:
Nit: match the above variable's docstring, since the names match.

`total aggregated sum of all weighted possible values`